src/main/java/com/diffmin/patch/PatchGeneration.java

```diff
@@ -83,8 +82,8 @@ private static CtElement delete(CtElement removedNode) {
                     @Override
                     public int getSrcNodeIndex(CtElement srcNode) {
                         CtElement srcNodeParent = srcNode.getParent();
-                        if (srcNodeParent.getValueByRole(srcNode.getRoleInParent())
-                                instanceof List) {
+                        if (srcNodeParent.getValueByRole(srcNode.getRoleInParent()) instanceof List
+                                || srcNode.getRoleInParent().equals(CtRole.CONTAINED_TYPE)) {
```

Comment:
I added this or condition to handle insertions only in `CtCompilationUnit` as I wanted to take advantage of the lists it uses to store the types. In `CtPackage`, I directly set all types.

Reply:
Don't use `.equals` for enums, use `==`. It has the same effect, _and_ is null safe.